tests/DeprecationTest.php

```diff
@@ -99,6 +80,7 @@ public function testDeprecation(): void
         );
 
         $this->assertEquals(2, Deprecation::getUniqueTriggeredDeprecationsCount());
+        restore_error_handler();
```

Comment:
This won't restore the error handler if the test fails. We should wrap everything that runs with a custom error handler into a try/finally block.